source/common/router/upstream_request.cc

```diff
@@ -567,22 +567,9 @@ void UpstreamRequest::onPoolReady(std::unique_ptr<GenericUpstream>&& upstream,
   // Have the upstream use the account of the downstream.
   upstream_->setAccount(parent_.callbacks()->account());
 
-  if (parent_.requestVcluster()) {
-    // The cluster increases its upstream_rq_total_ counter right before firing this onPoolReady
-    // callback. Hence, the upstream request increases the virtual cluster's upstream_rq_total_ stat
-    // here.
-    parent_.requestVcluster()->stats().upstream_rq_total_.inc();
-  }
-  if (parent_.routeStatsContext().has_value()) {
-    // The cluster increases its upstream_rq_total_ counter right before firing this onPoolReady
-    // callback. Hence, the upstream request increases the route level upstream_rq_total_ stat
-    // here.
-    parent_.routeStatsContext()->stats().upstream_rq_total_.inc();
-  }
```

Comment:
It looks like this block is moving to Filter::onUpstreamHostSelected, is that right? it's not super obvious to me that this isn't a behavior change with respect to when this method is called. But I'm sure you thought about it. Can you shed some light?

Reply:
AH! Thanks.